server/util.go

```diff
@@ -136,15 +136,30 @@ func externalIP() (string, error) {
 			case *net.IPAddr:
 				ip = v.IP
 			}
-			if ip == nil || ip.IsLoopback() {
-				continue
+			if external_ip := isPublicIP(ip); ip != nil {
+				return external_ip.String(), nil
 			}
-			ip = ip.To4()
-			if ip == nil {
-				continue // not an ipv4 address
-			}
-			return ip.String(), nil
 		}
 	}
 	return "", errors.New("are you connected to the network?")
 }
+
+// Taken from https://stackoverflow.com/questions/41670155/get-public-ip-in-golang
+func isPublicIP(IP net.IP) net.IP {
```

Comment:
Can we add support for IPv6?

Reply:
We can, it should involve changing `toIP4` to `toIP16`, and just doing a conditional check to ensure that its non-local, depending on whether ipv4 vs ipv6 is being used. We just need to find a table explaining the reserved parts of the IPv6 spectrum thats just used for local stuff.